### DISCUSSION TITLE
MySQL security group

### DISCUSSION THREAD

**Discussion by mflynn-lanl** [microbiomedata/nmdc-edge#269] [General]
@edwins Would it be possible to create a security group. The new security group would include the the MySQL VM and open port 3306. I would like to use this security group for the new MC cluster instead of MC creating a new one It would help connecting the MySQL VM to MC cluster.

**Comment by edwins** [comment]
It sounds like this security group would be added to the mysql host (to open 3306 on the mysql host), not to magic castle. Is that correct? If so, this security group probably shouldn't go into the mc template, and it should be scripted with a different template or even external script.

Security groups are attached to vms that need to be accessed or restricted.

**Comment by mflynn-lanl** [comment]
I would like both the mysql host and the mc cluster to both be in the security group so the login node can connect to the mysql host.
________________________________
From: Edwin Skidmore ***@***.***>
Sent: Monday, July 29, 2024 6:24 PM
To: microbiomedata/nmdc-edge ***@***.***>
Cc: Flynn, Mark C ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [microbiomedata/nmdc-edge] MySQL security group (Discussion #269)


It sounds like this security group would be added to the mysql host (to open 3306 on the mysql host), not to magic castle. Is that correct? If so, this security group probably shouldn't go into the mc template, and it should be scripted with a different template or even script.

**Reply by edwins** [reply]
Sorry, I edited my original response to provide a clarification, but it looks like you responded before seeing it. The security group applies to the mysql host based on your description of what needs to be accessed.

**Reply by edwins** [reply]
As an alternative, does accessing the mysql host using the private ip grant unrestricted access from the login node?

**Reply by edwins** [reply]
It might be the case that you're growing out of the standard Magic Castle template, and you might consider extending the template to adapt it to your own needs i.e. adding additional security rules etc. Do you have someone on your team with terraform experience? If so, this should be relatively easy to instruct them on how to do this and import that custom template into CACAO.

**Comment by mflynn-lanl** [comment]
I have the most terraform experience, lol
________________________________
From: Edwin Skidmore ***@***.***>
Sent: Tuesday, July 30, 2024 5:45 PM
To: microbiomedata/nmdc-edge ***@***.***>
Cc: Flynn, Mark C ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [microbiomedata/nmdc-edge] MySQL security group (Discussion #269)


It might be the case that you're growing out of the standard Magic Castle template, and you might consider extending the template to adapt it to your own needs i.e. adding additional security rules etc. Do you have someone on your team with terraform experience? If so, this should be relatively easy to instruct them on how to do this and import that custom template into CACAO.
